### DISCUSSION TITLE
declare property as const on initialization error, is it a bug?

### DISCUSSION THREAD

**Discussion by shemeshg** [hsutter/cppfront#1110] [Q&A]
I get error

```
note: non-static data member 'a' declared const here
```

for 

```
MyClass: type = {
	private a: const int ; // const cause a problem
	operator=: (
        out this,
        a_in: int
		) == {
			a = a_in;
		}
}
```

complete example

```
class MyClass2 {

    const int a;
	public:
    MyClass2(int a):a{a}{

    }
};

MyClass: type = {
	private a: const int ;
	operator=: (
        out this,
        a_in: int
		) == {
			a = a_in;
		}
}

hello: (msg: std::string_view) = {
   
    v: MyClass= MyClass(2);
    i :MyClass2 = MyClass2(2);
}
```

**Comment by zaucy** [comment]
```
operator:= (out this, a: int)
```

This lowers to a constructor **and** assignment so its not too surprising that would fail to compile. You can't re-assign const members in C++.

AFAIK there is no way to **only** define a constructor in cpp2.

**Reply by shemeshg** [reply]
So why is this works?

```
class MyClass2 {

    const int a;
	public:
    MyClass2(int a):a{a}{

    }
};
```

in cpp (none cpp2) it knows it will get it's value from the constructor so it wil no complain for const value not assigned.
( this is why I've pasted my complete test.h2 file, to show that c++ allows it)

**Reply by shemeshg** [reply]
Sorry, I think I understand what you meant, @zaucy. So, I should ask: 
Should this be lowered to a constructor and later assignment, and not (assignment only)? 
Or is it the desired behavior?
